internal/controller/serving/reconcilers/model_registry_reconciler.go

```diff
@@ -23,18 +28,31 @@ func NewModelRegistryInferenceServiceReconciler(client client.Client, log logr.L
 
 	err := client.List(context.Background(), &dscList)
 	if err != nil {
-		log.Error(err, "Failed to list DataScienceCluster")
+		return nil, fmt.Errorf("%w: %w", errGetDSC, err)
+	}
+
+	if len(dscList.Items) == 0 || len(dscList.Items) > 1 {
```

Comment:
Doesn't the opendatahub-operator already checks it?

Reply:
yes it does, for reference of this change a link to the discussion with @zdtsw https://github.com/opendatahub-io/odh-model-controller/pull/369#discussion_r1961814344